### PR TITLE
ECDSA sortition pool registration 

### PR DESCRIPTION
**Pull request by tomaszslabon**
Closes https://github.com/keep-network/keep-core/issues/3037 
Closes https://github.com/keep-network/keep-core/issues/3038
Closes https://github.com/keep-network/keep-core/issues/3039

Ideally it should only be merged to main after the `WalletRegistry` contract deployment is added to local installation scripts,
since the `WalletRegistry` is accessed (or the code accessing `WalletRegistry` and `SortitionPool` should be commented out for the time being).

This PR enables registration of the client in ECDSA sortition pool.
1) Added sortition pool contract handle to Tbtc chain.
2) Implemented functions from `sortition.Chain` in Tbtc chain.

The three checks (sortition pool registration, state update and reward eligibility update) are done in  `MonitorPool`.

### CODE REVIEW

**Comment by dimpar** [comment]
no comments, looks good to me

File: pkg/tbtc/tbtc.go [hunk 1]
```diff
@@ -23,14 +25,13 @@ func Initialize(
...
 
-	// TODO: Uncomment once https://github.com/keep-network/keep-core/pull/3094 is merged.
-	// err := sortition.MonitorPool(ctx, chain, sortition.DefaultStatusCheckTick)
-	// if err != nil {
-	// 	return fmt.Errorf(
-	// 		"could not set up sortition pool monitoring: [%v]",
-	// 		err,
-	// 	)
-	// }
+	err := sortition.MonitorPool(ctx, chain, sortition.DefaultStatusCheckTick)
```
Comment (pdyraga) [inline pkg/tbtc/tbtc.go:28]: I haven't had a chance to test it but how does the logs from the monitoring loop look like? Are we not having the same logs duplicated for Random Beacon and TBTC? Is there any way for the client operator to distinguish which sortition pool is up to date? See [this](https://github.com/keep-network/keep-core/pull/3093#discussion_r928722939) conversation please.

Comment (tomaszslabon) [reply pkg/tbtc/tbtc.go:28]: It seems the logs from monitoring Random Beacon and TBTC are indeed duplicated now, e.g.:
```
2022-07-26T11:11:01.422+0200    INFO    keep-sortition   sortition/sortition.go:65 checking sortition pool operator status
```

and then later:
```
2022-07-26T11:11:01.441+0200    INFO    keep-sortition   sortition/sortition.go:65 checking sortition pool operator status
```

**Comment by pdyraga** [comment]
👀

**Comment by pdyraga** [comment]
Tested locally, works as expected:
```
2022-07-27T11:55:22.356+0200	INFO	keep-beacon	sortition/sortition.go:64	checking sortition pool operator status
2022-07-27T11:55:22.357+0200	INFO	keep-beacon	sortition/sortition.go:77	operator is in the sortition pool
2022-07-27T11:55:22.357+0200	INFO	keep-beacon	sortition/sortition.go:131	operator is eligible for rewards
2022-07-27T11:55:22.357+0200	INFO	keep-beacon	sortition/sortition.go:89	sortition pool operator weight is up to date
2022-07-27T11:55:22.357+0200	INFO	keep-tbtc	sortition/sortition.go:64	checking sortition pool operator status
2022-07-27T11:55:22.358+0200	INFO	keep-tbtc	sortition/sortition.go:77	operator is in the sortition pool
2022-07-27T11:55:22.359+0200	INFO	keep-tbtc	sortition/sortition.go:131	operator is eligible for rewards
2022-07-27T11:55:22.359+0200	INFO	keep-tbtc	sortition/sortition.go:89	sortition pool operator weight is up to date
```

**Review by pdyraga — APPROVED** [review]

**Pull request merged**